flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -902,6 +905,28 @@ public CompletableFuture<String> triggerCheckpoint(JobID jobID, Time timeout) {
                 jobID, gateway -> gateway.triggerCheckpoint(timeout));
     }
 
+    @Override
+    public CompletableFuture<Acknowledge> triggerCheckpoint(
+            AsynchronousJobOperationKey operationKey, CheckpointType checkpointType, Time timeout) {
+        return dispatcherCachedOperationsHandler.triggerCheckpoint(
+                operationKey, checkpointType, timeout);
+    }
+
+    @Override
+    public CompletableFuture<OperationResult<Long>> getTriggeredCheckpointStatus(
+            AsynchronousJobOperationKey operationKey) {
+        return dispatcherCachedOperationsHandler.getCheckpointStatus(operationKey);
+    }
+
+    public CompletableFuture<Long> triggerCheckpointAndGetCheckpointID(
```

Comment:
Good catch. Was just following the [corresponding savepoint function](https://github.com/apache/flink/blob/9152af41c2d401e5eacddee1bb10d1b6bea6c61a/flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java#L916-L931). Just realized this could be unnecessary since this function is not needed in [EmbeddedJobClient](https://github.com/apache/flink/blob/cd8ea8d5b207569f68acc5a3c8db95cd2ca47ba6/flink-clients/src/main/java/org/apache/flink/client/deployment/application/EmbeddedJobClient.java)

Reply:
Will make it private